spot-contracts/contracts/FeePolicy.sol

```diff
@@ -0,0 +1,259 @@
+        perpRolloverFeeAPR.upper = p.upper;
+        perpRolloverFeeAPR.growth = p.growth;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpMintFeePerc() external override returns (uint256) {
+        return isOverSubscribed() ? 0 : maxPerpMintFeePerc;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpBurnFeePerc() external override returns (uint256) {
+        return isOverSubscribed() ? maxPerpBurnFeePerc : 0;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpRolloverFeePerc() external override returns (int256) {
+        IBondIssuer bondIssuer = perp.bondIssuer();
+        SubscriptionState memory s = computeSubscriptionState(bondIssuer);
+
+        int256 rolloverAPR = Sigmoid.compute(
+            s.normalizedSubscriptionRatio.toInt256(),
+            perpRolloverFeeAPR.lower,
+            perpRolloverFeeAPR.upper,
+            perpRolloverFeeAPR.growth,
+            ONE.toInt256()
+        );
+
+        // We calculate the rollover fee for the given cycle by dividing the annualized rate
+        // by the number of cycles in any given year.
+        return (rolloverAPR * bondIssuer.maxMaturityDuration().toInt256()) / ONE_YEAR_SEC;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computeVaultMintFeePerc() external view override returns (uint256) {
+        return maxVaultMintFeePerc;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computeVaultBurnFeePerc() external view override returns (uint256) {
+        return maxVaultBurnFeePerc;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computeVaultDeploymentFee() external view override returns (uint256) {
+        return vaultDeploymentFee;
+    }
+
+    /// @inheritdoc IFeePolicy
+    function decimals() external pure override returns (uint8) {
+        return DECIMALS;
+    }
+
+    /// @return If the rollover vault is over-subscribed based on the current tvls.
+    function isOverSubscribed() public returns (bool) {
+        SubscriptionState memory s = computeSubscriptionState(perp.bondIssuer());
+        return s.normalizedSubscriptionRatio > ONE;
+    }
+
+    /// @notice Computes the subscription detailed state based on the current `currentVaultTVL` and `targetVaultTVL`.
+    /// @return The subscription state.
+    function computeSubscriptionState(IBondIssuer bondIssuer) public returns (SubscriptionState memory) {
+        SubscriptionState memory s;
+        s.currentPerpTVL = perp.getTVL();
+        s.currentVaultTVL = vault.getTVL();
+        // NOTE: We assume that perp minting bonds have only 2 tranches and perp assumes the senior one.
+        //       We assume that perp's TVL and vault's TVL values have the same base denomination.
+        s.targetVaultTVL = s
+            .currentPerpTVL
+            .mulDiv(bondIssuer.trancheRatios(1), bondIssuer.trancheRatios(0), MathUpgradeable.Rounding.Up)
+            .mulDiv(ONE, targetSubscriptionRatio, MathUpgradeable.Rounding.Up);
```

Comment:
```suggestion
            .mulDiv(targetSubscriptionRatio, ONE, MathUpgradeable.Rounding.Up);
```

Reply:
What does this line do? I understand up until here. Naming updates should make this much easier to follow.